### PR TITLE
OP-24: thresholds.py — every tunable in one injected frozen dataclass

### PR DESCRIPTION
**Pull request by m-nweke**
Closes OP-24. Base: `op-23-geometry`.

Collects every tunable into one frozen dataclass, injected rather than global.

## Changes
- `posture_core.thresholds`: `Thresholds` (17 fields), `DEFAULT_THRESHOLDS`, `RULES_VERSION`.

## Notes
- Replaces constants scattered through the code that used them, including the `±100` pixel arm-crossing literal whose own comment said it "shall be replaced with a calculation which can adjust to different sizes of people".
- Injected, not global: boundary tests construct their own instance rather than monkeypatching, so a test cannot leak into the next one.
- No loading here. `posture_core` reads no files or environment variables; the API layer loads and passes down (OP-39), and OP-36 moves the defaults to `posture-spec/rules.json`.
- `__post_init__` rejects contradictory orderings — upright above slouch, positive recline, inverted knee band, view ratios with a negative gap. These produce a band no value can fall into, so the rule silently never fires.
- `cva_forward_head_deg` must be below `cva_borderline_deg`: a smaller craniovertebral angle is worse, the opposite of every other threshold here.
- Defaults are starting points, clinical where a standard exists. `arms_crossed_ratio` is revised against measurements in OP-28.

## Tests
12 tests, 100% coverage, mostly on the validation.

### CODE REVIEW

File: packages/posture-core/tests/test_thresholds.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    with pytest.raises(dataclasses.FrozenInstanceError):
+        DEFAULT_THRESHOLDS.trunk_slouch_deg = 25.0  # type: ignore[misc]
+
+
+def test_a_variant_is_made_by_replacement_not_mutation() -> None:
+    """The pattern every boundary test uses."""
+    strict = dataclasses.replace(DEFAULT_THRESHOLDS, trunk_slouch_deg=15.0)
+    assert strict.trunk_slouch_deg == 15.0
+    assert DEFAULT_THRESHOLDS.trunk_slouch_deg == 20.0
+    assert strict.min_visibility == DEFAULT_THRESHOLDS.min_visibility
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/tests/test_thresholds.py]: Fixed. The test now captures `DEFAULT_THRESHOLDS.trunk_slouch_deg` and asserts the shared instance still holds it, rather than pinning 20.0.

You are right about the coupling: this module explicitly documents the defaults as expected to be retuned, so a literal here made every retune a two-file edit while adding no protection. What the test is actually for is that `dataclasses.replace` does not mutate the shared instance, and that is now what it says.

File: packages/posture-core/src/posture_core/thresholds.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+            raise ValueError(
+                f"trunk_upright_deg ({self.trunk_upright_deg}) must not exceed "
+                f"trunk_slouch_deg ({self.trunk_slouch_deg}); nothing could fall between them"
+            )
+        if self.trunk_recline_deg >= 0.0:
+            raise ValueError(
+                f"trunk_recline_deg is a backward lean and must be negative, "
+                f"got {self.trunk_recline_deg}"
+            )
+        if self.cva_forward_head_deg > self.cva_borderline_deg:
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/src/posture_core/thresholds.py]: Fixed — the comparison is now `>=` and the message says "strictly below".

Equality is not a harmless edge case, and it fails in exactly the way this validation exists to catch. With both at 50, `value < 50` claims the major finding and the identical borderline test can never be reached, so `forward_head_borderline` becomes unreachable — a silently dead rule, no exception, no bad number. Added a test for the equal case.

File: packages/posture-core/src/posture_core/synthetic.py [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+    """World metres -> canonical landmarks carrying both coordinate systems.
+
+    Orthographic: the depth axis is dropped rather than divided through. A real camera's
+    perspective divide matters when the subject's depth is comparable to their distance from the
+    lens, which is not the case for someone photographed at desk range — and modelling it would
+    make the figure's *known* angles no longer recoverable from the image, defeating the purpose.
+    """
+    scale = image_height * _FRAME_HEIGHTS_PER_METRE
+    origin_x = image_width * _HIP_X_FRACTION
+    origin_y = image_height * _HIP_Y_FRACTION
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/src/posture_core/synthetic.py:376]: Confirmed — `make_pose(image_width=0)` raises `ZeroDivisionError: float division by zero`, not the intended message.

`synthetic.py` shipped in #26 and is already merged, so this cannot be fixed in this PR. Tracked as **OP-62** with the other robustness findings from this review round and fixed in a follow-up PR against `main`.

File: packages/pose-backends/src/pose_backends/registry.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    9 MB, heavy at 29 MB. Nothing in the code depends on which one loads — all three emit the same
+    33 landmarks with world coordinates — so this is a configuration change, not a code change.
+    See ``models/checksums.txt`` for the pins and the tradeoffs.
+
+    It also replaces the legacy arrangement's worst property: weights fetched from a bare Dropbox
+    link recorded in a readme, with no checksum and no stated origin, so a dead link made the whole
+    project unrunnable and a corrupted copy was indistinguishable from a good one.
+    """
+    override = os.environ.get("MODEL_PATH")
+    return Path(override) if override else DEFAULT_MODEL_PATH
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/pose-backends/src/pose_backends/registry.py:53]: Half confirmed, and the half that holds is worth fixing.

Empty string is already handled: `Path(override) if override else DEFAULT_MODEL_PATH` treats `''` as falsy, so `MODEL_PATH=''` falls back to the default. Whitespace does not — `MODEL_PATH='   '` yields `PosixPath('   ')`, which is precisely the confusing downstream failure you describe.

`registry.py` is merged (#26/#27), so this goes into **OP-62** and its follow-up PR rather than here.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+	mv "$$tmp" "$(MODEL_TARGET)"; \
+	echo "OK  $(MODEL_TARGET)  $$actual"
+
+## Verify whatever is on disk. model-validation.yml runs this before anything else, so a corrupted
+## or swapped model fails the workflow instead of quietly changing its results.
+verify-model:
+	@expected="$(expected_sha)"; \
+	if [ ! -f "$(MODEL_TARGET)" ]; then \
+		echo "ERROR: $(MODEL_TARGET) is missing. Run \`make fetch-model\`." >&2; exit 1; \
+	fi; \
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply Makefile:78]: Confirmed by reproducing it. With a present-but-unpinned variant the output is:

```
ERROR: models/pose_landmarker_giant.task does not match its pin.
       expected
       actual   708811cc...
```

A blank `expected` reported as a mismatch, exactly as you describe — the real problem is that no pin exists. The `-z` guard from `fetch-model` belongs here too.

Note the missing-file check fires first, so this needs a file present but unpinned to reach. The Makefile is merged (#27), so this goes into **OP-62**.

File: packages/posture-core/src/posture_core/thresholds.py [hunk 6]
```diff
@@ -0,0 +1,189 @@
+"""Every tunable number in the rules engine, in one frozen, injected object.
+
+## Why this exists at all
+
+The inherited engine scattered its constants through the code it used them in — a `±100` pixel
```
[the remark this answers is not in the document]
Comment (m-nweke) [reply packages/posture-core/src/posture_core/thresholds.py:5]: Same display artifact as the equivalent comment on #30, not a scope problem.

This PR was retargeted to `main` while the branches below it were unmerged, so GitHub cached a file list computed against an older `main`. `git diff origin/main..origin/op-24-thresholds` is `thresholds.py` and `test_thresholds.py`. The geometry primitives are #30, and the synthetic builder, fake backend, CLI and model tooling landed in #26 and #27, already merged.

The count collapses as the stack merges; the merge uses the true merge-base regardless. Description left as-is because it describes the commit.
